### PR TITLE
feat(alert): #MA-967 add of the count of alerts per line

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
Party 1/3 of the ticket [MA-770](https://entsupport.gdapublic.fr/browse/MA-770)

Preparation of the "alert" and "alert_history" tables for the MA-770.
MA-770 aims to make alerts resettable by dates.

The tests are provided in the PR #140 

Doc: https://entconf.gdapublic.fr/pages/viewpage.action?pageId=96436457

The files in the "SQLReverseScript" folder is used in case of a problem in production. They must allow to restore the database to the state before the ticket. They will be deleted in the PR of the MA-770 ticket.

## Checklist tests
Nothing. Just execute this file after MA-662 complete 
https://github.com/OPEN-ENT-NG/presences/pull/140/files#diff-849bfe6ffcfd3d258ea47768804e674f38c73d03c0da7520bebbcbef54fbccd2

## Issue ticket number and link
[MA-967](https://entsupport.gdapublic.fr/browse/MA-967)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+-- Delete old data due to the modification of the table
+DELETE FROM presences.alerts WHERE 1 = 1;
```
Comment (Sofianebernoussi) [inline presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql]: tu peux juste utiliser truncate table no?

Comment (JinDouang) [reply presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql]: Yes
Et si tu veux juste delete, pas besoin de la conditionner à ma connaissance ?

File: presences/src/main/resources/sql/scriptSQLReverse/044-MA-967-count-alert-by-line.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+-- Delete old data due to the modification of the table
```
Comment (JinDouang) [inline presences/src/main/resources/sql/scriptSQLReverse/044-MA-967-count-alert-by-line.sql]: Tu as 2 fichiers SQL, tu peux changer leur nommages en fonction de ce que tu souhaites ?

File: presences/src/main/resources/sql/scriptSQLReverse/044-MA-967-count-alert-by-line.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+DELETE FROM presences.alerts WHERE 1 = 1;
+DELETE FROM presences.alert_history WHERE 1 = 1;
```
Comment (JinDouang) [inline presences/src/main/resources/sql/scriptSQLReverse/044-MA-967-count-alert-by-line.sql]: Idem pour truncate/delete sans condition

**Review by JinDouang — CHANGES_REQUESTED** [review]

File: presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql (hunk 1 above)
Comment (qpic-cgi) [reply presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql]: yes, mais tu te prend un warning en mode "attention y'a pas de condition donc tu vas tout supprimer"

Comment (Sofianebernoussi) [reply presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql]: The WHERE clause specifies which record(s) should be deleted. If you omit the WHERE clause, all records in the table will be deleted!

Donc non t'as raison Jin !

**Review by flmartineau — APPROVED** [review]

File: presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql (hunk 1 above)
Comment (qpic-cgi) [reply presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql]: ouai cette phrase là

Comment (BastienBenard) [reply presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql]: DELETE FROM presences.alerts ☑
TRUNCATE TABLE presences.alerts ☑

File: presences/src/main/resources/sql/scriptSQLReverse/044-MA-967-count-alert-by-line.sql (hunk 3 above)
Comment (BastienBenard) [reply presences/src/main/resources/sql/scriptSQLReverse/044-MA-967-count-alert-by-line.sql]: DELETE FROM presences.alerts ☑
TRUNCATE TABLE presences.alerts ☑

File: presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql (hunk 1 above)
Comment (JinDouang) [reply presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql]: > yes, mais tu te prend un warning en mode "attention y'a pas de condition donc tu vas tout supprimer"

Mais du coup c'est ce que Bastien veut, non ?

**Review by JinDouang — APPROVED** [review]

File: presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+TRUNCATE TABLE presences.alert_history;
+
+-- Override unique CONSTRAINT
+ALTER TABLE presences.alerts ADD COLUMN event_id bigint NOT NULL;
+ALTER TABLE presences.alerts DROP CONSTRAINT uniq_alert;
+ALTER TABLE presences.alerts ADD CONSTRAINT uniq_alert UNIQUE (event_id, structure_id, student_id, type);
+
+
+-- Delete column no longer used
+ALTER TABLE presences.alerts DROP COLUMN exceed_date;
```
Comment (Sofianebernoussi) [inline presences/src/main/resources/sql/044-MA-967-count-alert-by-line.sql]: ALTER TABLE table_name
ADD [COLUMN] column_definition,
ADD [COLUMN] column_definition,

Pareil pour drop pour éviter de réecrire les alter table a voir si ca défonce pas la lisibilité

**Review by qpic-cgi — APPROVED** [review]

**Pull request merged**